src/api/firebase.js

```diff
@@ -193,10 +199,37 @@ export async function updateItem(listPath, item) {
 	const listCollectionRef = collection(db, listPath, 'items');
 	const itemRef = doc(listCollectionRef, item.id);
 
+	const currentDate = Timestamp.now();
+	const lastUpdate = item.dateLastPurchased || item.dateCreated;
+
+	// last estimated purchase interval
+	const previousEstimate = getDaysBetweenDates(
+		lastUpdate,
+		item.dateNextPurchased,
+	);
+
+	// number of days since the item was added to the list or last purchased
+	const daysSinceLastPurchase = getDaysBetweenDates(lastUpdate, currentDate);
+
+	// calculated estimate for the number of days until the next purchase
+	const daysUntilNextPurchase = calculateEstimate(
+		previousEstimate,
+		daysSinceLastPurchase,
+		item.totalPurchases,
+	);
+
+	// value for dateNextPurchased property
```

Comment:
nit: there is a syntax that allows multi-line comments, since we have here 3 lines I would recommend to use it: 
```
/*
value for dateNextPurchased property calculated by multiplying 24 hours(in millisecs) 
by the daysUntilNextPurchase, then adding to the current date
*/
```